addons/website/static/src/systray_items/edit_in_backend.js

```diff
@@ -1,29 +1,25 @@
 /** @odoo-module **/
 
 import { registry } from "@web/core/registry";
-import { useService } from "@web/core/utils/hooks";
-import { Dropdown } from "@web/core/dropdown/dropdown";
-import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { useService, useBus } from "@web/core/utils/hooks";
 
-const { Component, onMounted } = owl;
+const { Component, onWillStart, onMounted } = owl;
+
+const websiteSystrayRegistry = registry.category('website_systray');
 
 export class EditInBackendSystray extends Component {
     setup() {
         this.websiteService = useService('website');
         this.actionService = useService('action');
 
+        onWillStart(this._updateMainObjectName);
+        useBus(websiteSystrayRegistry, 'CONTENT-UPDATED', this._updateMainObjectName);
```

Comment:
Hello, the onWillStart will not be called again when switching between two pages which have a main object, because the component is not unmounted when doing that (and it has no props or state so it is not supposed to rerender).
The `CONTENT-UPDATED` event is sent when there is a new pageDocument so that the navbar rerenders. When the previous and the current pageDocument have both the editableInBackend metadata, the systray item will remain as displayed in the website systray registry, and onWillUnmount and onWillStart will not be called as nothing changed.
But when the new pageDocument has not that metadata, the navbar will rerender without that systray item and it will be unmounted, so on the next pageDocument with editableInBackend it will we mounted again and onWillStart will be called.

I think it would be better to define the mainObjectName variable as a state variable, as we want to rerender the component when it changes (I have to admit I don't understand why the component is rerendered with the correct mainObjectName right now, the `CONTENT-UPDATED` should only update the variable, so something else triggers a rerender of the component, but I don't think it is "intentional", as the component has no state nor props, so it may change if it's coming from a bug and it gets fixed. It is visible with a slower `getUserModelName` method).

Reply:
> I have to admit I don't understand why the component is rerendered with the correct mainObjectName right now

Yes, that's what I did not understand. But I guess we do not have to understand and just use a state variable indeed. @Guillaume-gdi would you handle that?

Reply:
Yes I'll handle it :+1:

Reply:
@qsm-odoo Here it is: https://github.com/odoo/odoo/pull/99599